app/build.gradle

```diff
@@ -77,6 +77,8 @@ dependencies {
     implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0'
     implementation 'androidx.vectordrawable:vectordrawable:1.2.0'
```

Comment:
As is, this PR is too big for me to accept. The refactoring of the messages look cool, but it IMO doesn't really fix something with the previous implementation, it's just a different implementation that uses the StateFlow stuff.

I'm not against refactoring, but it's basically a switch from battle tested code, that's running like this for multiple years to something that will have bugs that will fall back onto me.

Reply:
The new Implementation provides multiple things:
- A single SourceOfTruth where the data is stored/obtained
- A reactive approach that updates the UI in real time whenever any data is updated
- A clean separation of responsibility (eg: In the previous one, the UI wasn't only used to render the UI but even parse the API data that was received from the WS Connection)
- A Real MVVM implementation that allows the UI to subscribe to new data from the VM and the VM to be agnostic to what the View does.

I know the PR is a bit bigger, and just because I knew it, I split it into multiple commits that can be merged 1 by 1, keeping the app working as the previous commit worked.
If you want, I can split the PR into smaller PRs If it helps you on the review process